lib/dom.js

```diff
@@ -1060,6 +1068,167 @@ Node.prototype = {
+	 *
+	 * @type {string}
+	 */
+	baseURI: 'about:blank',
+	/**
+	 * Is true if this node is part of a document.
+	 *
+	 * @type {boolean}
+	 */
+	get isConnected() {
+		var rootNode = this.getRootNode();
+		if (rootNode && rootNode.nodeType === rootNode.DOCUMENT_NODE) {
+			return true;
+		}
+
+		return false;
+	},
+	/**
+	 * Checks whether or not `other` is an inclusive descendant of this node.
+	 *
+	 * @param {Node} [other]
+	 * The node to check.
+	 * @returns {boolean}
+	 * True if `other` is an inclusive descendant of this node; false otherwise.
+	 * @see https://dom.spec.whatwg.org/#dom-node-contains
+	 */
+	contains: function (other) {
+		if (!other) {
+			return false;
+		}
+
+		do {
+			if (this === other) {
+				return true;
+			}
+		} while ((other = other.parentNode));
+
+		return false;
+	},
+	/**
+	 * @typedef GetRootNodeOptions
+	 * @property {boolean} [composed=false]
+	 */
+	/**
+	 * Searches for the root node of this node.
+	 *
+	 * Difference to the specs:
+	 * options.compose doesn't change this behaviour.
+	 *
+	 * @param {GetRootNodeOptions} [options]
+	 * @returns {Node}
+	 * Root node.
+	 * @see https://dom.spec.whatwg.org/#dom-node-getrootnode
+	 */
+	getRootNode: function (options) {
+		if (options && options.composed) {
+			// should return shadow root here (not supported)
+		}
+
+		var parent = this;
+		do {
+			if (!parent.parentNode) {
+				return parent;
+			}
+		} while ((parent = parent.parentNode));
+	},
+	/**
+	 * Checks whether or not the given node is equal to this node.
+	 *
+	 * @param {Node} [otherNode]
+	 * @see https://dom.spec.whatwg.org/#concept-node-equals
+	 */
+	isEqualNode: function (otherNode) {
+		if (!otherNode) {
+			return false;
+		}
+
+		if (this.nodeType !== otherNode.nodeType) {
+			return false;
+		}
+
+		if (
+			(this.childNodes && otherNode.childNodes && this.childNodes.length !== otherNode.childNodes.length) ||
+			(!this.childNodes && otherNode.childNodes) ||
+			(this.childNodes && !otherNode.childNodes)
+		) {
+			return false;
+		}
+
+		if (this.childNodes && otherNode.childNodes) {
+			for (var i = 0; i < this.childNodes.length; i++) {
+				if (!this.childNodes[i].isEqualNode(otherNode.childNodes[i])) {
+					return false;
+				}
+			}
+		}
+
+		switch (this.nodeType) {
+			case this.DOCUMENT_TYPE_NODE:
+				if (this.name === otherNode.name && this.publicId === otherNode.publicId && this.systemId === otherNode.systemId) {
+					return true;
+				}
+				break;
+			case this.ELEMENT_NODE:
+				if (
+					this.namespaceURI !== otherNode.namespaceURI ||
+					this.prefix !== otherNode.prefix ||
+					this.localName !== otherNode.localName ||
+					(this.attributes && otherNode.attributes && this.attributes.length !== otherNode.attributes.length) ||
+					(!this.attributes && otherNode.attributes) ||
+					(this.attributes && !otherNode.attributes)
+				) {
+					return false;
+				}
+
+				if (this.attributes && otherNode.attributes) {
```

Comment:
But I think for `Elements`s we know that attributed can not be falsy, so here it should be possible to simplify the conditions.
I would even recommend to split the first condition into multiple short if conditions that each return `false`,
the last one being
`this.attributes.length !== otherNode.attributes.length`.
The one around the for loop should not be needed at all.